src/core/AudioResampler.cpp

```diff
@@ -26,44 +26,84 @@
 
 #include <samplerate.h>
 #include <stdexcept>
-#include <string>
+#include <utility>
 
 namespace lmms {
 
-AudioResampler::AudioResampler(int interpolationMode, int channels)
-	: m_interpolationMode(interpolationMode)
+AudioResampler::AudioResampler(Mode mode, ch_cnt_t channels)
+	: m_state(src_new(converterType(mode), channels, &m_error))
+	, m_mode(mode)
 	, m_channels(channels)
-	, m_state(src_new(interpolationMode, channels, &m_error))
 {
-	if (!m_state)
-	{
-		const auto errorMessage = std::string{src_strerror(m_error)};
-		const auto fullMessage = std::string{"Failed to create an AudioResampler: "} + errorMessage;
-		throw std::runtime_error{fullMessage};
-	}
+	if (channels <= 0) { throw std::logic_error{"Invalid channel count"}; }
+	if (!m_state) { throw std::runtime_error{src_strerror(m_error)}; }
 }
 
 AudioResampler::~AudioResampler()
 {
 	src_delete(m_state);
 }
 
-auto AudioResampler::resample(const float* in, long inputFrames, float* out, long outputFrames, double ratio)
-	-> ProcessResult
+AudioResampler::AudioResampler(AudioResampler&& other) noexcept
```

Comment:
`m_ratio` and `m_error` are not handled. Same goes for the move assignment operator.